extensions/okhttp/src/main/java/com/google/android/exoplayer2/ext/okhttp/OkHttpDataSource.java

```diff
@@ -280,7 +276,6 @@ private Request makeRequest(DataSpec dataSpec) {
       }
       builder.addHeader("Range", rangeRequest);
     }
-    builder.addHeader("User-Agent", userAgent);
```

Comment:
Why not just add a null check here, remove the assertion from the constructor, and update the Javadoc to indicate that null is allowed? It seems unnecessary to remove the functionality, which is useful if you want to use one userAgent for requests made by the player, and another for requests made elsewhere by your app.